eureka2-client/src/main/java/com/netflix/eureka2/client/interest/BatchingRegistryImpl.java

```diff
@@ -158,12 +152,9 @@ public void shutdown() {
 
     private BufferState shouldBatchAtomic(Interest<T> atomic) {
         BufferState state = interestsBufferingState.get(atomic);
-        if (state == null) {
```

Comment:
maybe use a switch(state) here? That way we can better express the state machine.